src/test/java/org/forgerock/cuppa/CuppaTest.java

```diff
@@ -299,6 +299,134 @@ public void basicApiUsageWithPassingFailingAndErroredTests() {
         assertTestResources(results, 1, 1, 1);
     }
 
+    @Test
+    public void beforeShouldRunOnceBeforeTests() {
+
+        //Given
+        AtomicInteger hasTopLevelBeforeFunctionRun = new AtomicInteger();
```

Comment:
Sure, but if we have more than one method its not a functional interface. We could instead just expose the ```callCount``` variable.